ci/include-grouping.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+"""
+easy command line to run against all citus-style checked files:
+
+$ git ls-files \
+  | git check-attr --stdin citus-style \
+  | grep 'citus-style: set' \
+  | awk '{print $1}' \
+  | cut -d':' -f1 \
+  | xargs -n1 ./ci/include-grouping.py
+"""
+
+
+import os
+import sys
+
+
+def main(args):
+    if len(args) < 2:
+        print("Usage: include-grouping.py <file>")
+        return
+
+    file = args[1]
+    if not os.path.isfile(file):
+        print("File does not exist", file=sys.stderr)
+        return sys.exit(1)
+
+    with open(file, "r") as f:
+        with open(file + ".tmp", "w") as out_file:
+            lines = f.readlines()
+            includes = []
+            skipped_lines = []
+            for line in lines:
+                if line.startswith("#include"):
+                    includes.append(line)
+                    skipped_lines = []
+                else:
+                    if len(includes) > 0:
```

Comment:
I think this loop is deserving of a comment of what it does. Something like:
> This calls print_sorted_includes on a set of consecutive #include lines. This implicitly keeps separation of any #include lines that are contained in an #ifdef, because it will order the #include lines inside and after the #ifdef completely separately.